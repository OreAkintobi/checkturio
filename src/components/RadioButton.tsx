import React from 'react';
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';
import { theme } from '../utils';

interface BooleanInputProps {
  value: string;
  isSelected?: boolean;
  onSelect: () => void;
}

export const RadioButton: React.FC<BooleanInputProps> = ({
  value,
  isSelected,
  onSelect,
}) => {
  return (
    <TouchableOpacity onPress={onSelect}>
      <View style={styles.radioContainer}>
        <View style={styles.radioIndicator}>
          {isSelected ? <View style={styles.selectedIndicator} /> : null}
        </View>
        <Text style={styles.label}>{value}</Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  radioContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: theme.sizes.m,
    marginRight: theme.sizes.m,
    borderRadius: theme.sizes.s,
    padding: theme.sizes.m,
    borderColor: theme.colors.border,
    borderWidth: 1,
  },
  radioIndicator: {
    width: theme.sizes.l,
    height: theme.sizes.l,
    borderRadius: theme.sizes.l,
    backgroundColor: theme.colors.background,
    marginRight: theme.sizes.m,
    borderColor: theme.colors.border,
    borderWidth: 0.5,
    justifyContent: 'center',
    alignItems: 'center',
  },
  selectedIndicator: {
    width: theme.sizes.m,
    height: theme.sizes.m,
    borderRadius: theme.sizes.m,
    backgroundColor: theme.colors.primary,
  },
  label: {
    color: theme.colors.primary,
    fontSize: theme.sizes.xl,
  },
});
