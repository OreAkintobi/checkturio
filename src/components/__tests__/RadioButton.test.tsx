import React from 'react';
import renderer from 'react-test-renderer';
import { RadioButton } from '../RadioButton';

describe('<RadioButton />', () => {
  const onSelectMock = jest.fn();

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <RadioButton
          value="Option"
          isSelected={false}
          onSelect={onSelectMock}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onSelect when pressed', () => {
    const component = renderer.create(
      <RadioButton value="Option" isSelected={false} onSelect={onSelectMock} />
    );
    const instance = component.root;
    instance.props.onSelect(); // Directly call onSelect prop
    expect(onSelectMock).toHaveBeenCalledTimes(1);
  });

  it('displays selected indicator when isSelected is true', () => {
    const tree = renderer
      .create(
        <RadioButton value="Option" isSelected={true} onSelect={onSelectMock} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('does not display selected indicator when isSelected is false', () => {
    const tree = renderer
      .create(
        <RadioButton
          value="Option"
          isSelected={false}
          onSelect={onSelectMock}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
