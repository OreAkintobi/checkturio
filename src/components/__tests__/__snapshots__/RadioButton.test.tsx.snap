// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> displays selected indicator when isSelected is true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#ced4da",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 8,
        "marginRight": 8,
        "padding": 8,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderColor": "#ced4da",
          "borderRadius": 12,
          "borderWidth": 0.5,
          "height": 12,
          "justifyContent": "center",
          "marginRight": 8,
          "width": 12,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#007bff",
            "borderRadius": 8,
            "height": 8,
            "width": 8,
          }
        }
      />
    </View>
    <Text
      style={
        {
          "color": "#007bff",
          "fontSize": 16,
        }
      }
    >
      Option
    </Text>
  </View>
</View>
`;

exports[`<RadioButton /> does not display selected indicator when isSelected is false 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#ced4da",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 8,
        "marginRight": 8,
        "padding": 8,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderColor": "#ced4da",
          "borderRadius": 12,
          "borderWidth": 0.5,
          "height": 12,
          "justifyContent": "center",
          "marginRight": 8,
          "width": 12,
        }
      }
    />
    <Text
      style={
        {
          "color": "#007bff",
          "fontSize": 16,
        }
      }
    >
      Option
    </Text>
  </View>
</View>
`;

exports[`<RadioButton /> renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#ced4da",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 8,
        "marginRight": 8,
        "padding": 8,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderColor": "#ced4da",
          "borderRadius": 12,
          "borderWidth": 0.5,
          "height": 12,
          "justifyContent": "center",
          "marginRight": 8,
          "width": 12,
        }
      }
    />
    <Text
      style={
        {
          "color": "#007bff",
          "fontSize": 16,
        }
      }
    >
      Option
    </Text>
  </View>
</View>
`;
